zebra/rib.h

```diff
@@ -53,6 +53,13 @@ struct rnh {
 	struct prefix resolved_route;
 	struct list *client_list;
 
+	/* For the fully resolved route */
+	uint8_t resolved_type;
+	uint8_t resolved_distance;
+	uint16_t resolved_instance;
+	uint32_t resolved_metric;
+	struct prefix resolved_prefix;
```

Comment:
How is this different from `resolved_route`?

Reply:
The resolve_route is the one from the first lookup in zebra_rnh_resolve_nexthop_entry(). In case of a recursive nexthop, the resolved_route is different from the fully-resolved route.